### PR TITLE
Class static private field destructure set

### PR DESCRIPTION
**Pull request by JLHwung**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | Runtime error is thrown when running transpiled `[C.#p] = [0]` ([REPL](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=true&loose=false&code_lz=MYGwhgzhAEDC0G8BQ1oQC5nQS2NAxAA4DcK0wA9gHYYBOArsOhbQBQCUiZqA2rAHREAutAC80HgAYhpVAF8kCpFQCmAdzjEgA&debug=false&forceAllTransforms=false&shippedProposals=true&circleciRepo=&evaluate=true&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact&prettier=false&targets=&version=7.13.7&externalPlugins=))
| Patch: Bug Fix?          | Y
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      | Yes
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

This PR depends on #12910, please review that one first.

In this PR we introduce a new helper `classStaticPrivateFieldDestructureSet` similar to `classPrivateFieldDestructureSet` except that it accepts a classRef and descriptor. From this perspective this PR can be considered the `static` flavor of #10017.

### CODE REVIEW

File: packages/babel-helpers/src/helpers.js
Comment (jridgewell) [inline packages/babel-helpers/src/helpers.js:1420]: The more I look at this, I think we can actually reuse this code in the spec instance helpers. Eg, get the descriptor from the instance map, then call over to here. The only change necessary would be for `receiver !== classConstructor` to be changed a bit.

**Review by jridgewell — APPROVED** [review]

File: packages/babel-plugin-proposal-class-properties/test/fixtures/private/access-before-declaration/exec.js
Comment (JLHwung) [reply packages/babel-plugin-proposal-class-properties/test/fixtures/private/access-before-declaration/exec.js:1]: Yeah, this one is based on #12910, see https://github.com/babel/babel/pull/12917/files/dd660bce888e8e06a7e11430d796b8fa54a3ffd4..3a307137ac24195afb30acd18f240d03a48c6cd9 for the real diff.

File: packages/babel-helpers/src/helpers.js
Comment (JLHwung) [reply packages/babel-helpers/src/helpers.js:1420]: Agreed, I will come up with refactors later. Also we should check if `classStaticPrivateFieldDestructureSet` exists because of backward compatibility to older `@babel/helpers`.

File: packages/babel-plugin-proposal-private-methods/test/fixtures/private-static-method/read-only/output.js
Comment (jridgewell) [inline packages/babel-plugin-proposal-private-methods/test/fixtures/private-static-method/read-only/output.js:3]: In a followup, we need to make this better. We're passing `_method` where we expect `descriptor`.

File: Makefile
Comment (nicolo-ribaudo) [inline Makefile:209]: We also need to disable the release action manual trigger since it doesn't use this `new-version` command, just to be sure :grimacing:

File: packages/babel-plugin-proposal-private-methods/test/fixtures/private-static-method/read-only/output.js
Comment (nicolo-ribaudo) [reply packages/babel-plugin-proposal-private-methods/test/fixtures/private-static-method/read-only/output.js:3]: In Babel 8 I'd like to just remove the concept of descriptors for private elements.
I introduced them for decorators, but the decorators proposal has evolved to be more static and now we don't need runtime descriptors anymore.

**Review by nicolo-ribaudo — DISMISSED** [review]

**Comment by nicolo-ribaudo** [comment]
`Do not merge` until #12910 is merged.

File: packages/babel-helpers/src/helpers.js
Comment (nicolo-ribaudo) [inline packages/babel-helpers/src/helpers.js:1301]: We don't need to bump the version because it's functionally equivalent, right?

**Review by nicolo-ribaudo — APPROVED** [review]

File: packages/babel-helpers/src/helpers.js
Comment (JLHwung) [reply packages/babel-helpers/src/helpers.js:1301]: Yes. Refactoring internals helpers should not bump minimal required helper version as long as the interface is not changed.

**Pull request merged**
